### PR TITLE
fix(backend): add missing "/api" prefix for white route check bb-11

### PR DESCRIPTION
**Pull request by KeidsID**
Sorry guys 🙏🏽 .

### CODE REVIEW

File: apps/backend/src/libs/plugins/authorization-plugin/authorization-plugin.ts [hunk 1]
```diff
@@ -19,15 +19,17 @@ const authorizationPlugin = fp<PluginOptions>((app, options, done) => {
 	const { token, userService, whiteRoutes = [] } = options;
 
 	app.addHook(ServerHooks.PRE_HANDLER, async (request) => {
-		const isWhiteRoute = whiteRoutes.some((whiteRoute) =>
-			request.url.endsWith(whiteRoute),
-		);
+		const { headers, url } = request;
+
+		const isWhiteRoute = whiteRoutes.some((whiteRoute) => {
+			return url.startsWith("/api") && url.endsWith(whiteRoute);
```
Comment (fshabanov) [inline apps/backend/src/libs/plugins/authorization-plugin/authorization-plugin.ts:25]: Use regex here. The url should match /api/v{number/endpoint, take endpoint from there and check if it is in white routes. Create a helper for that

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const checkWhiteRoute = (route: string, whiteRoutes: string[]): boolean => {
+	return whiteRoutes.some((whiteRoute) => {
+		const regex = new RegExp(`^/api.*${whiteRoute}$`);
```
Comment (fshabanov) [inline apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts:3]: `^\/api\/v\d+.*(\/.+)$`
Then use `route.match(regex)` - the response will be an array, where the second element will be the endpoint (for example /sign-in)
Then check if that endpoint is in white route

File: apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+const checkWhiteRoute = (route: string, whiteRoutes: string[]): boolean => {
```
Comment (fshabanov) [inline apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts:1]:
```suggestion
const checkIsWhiteRoute = (route: string, whiteRoutes: string[]): boolean => {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts (hunk 3 above)
Comment (KeidsID) [reply apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts:1]: Do I need to rename the file into `check-is-white-route.helper.ts`?

Comment (fshabanov) [reply apps/backend/src/libs/plugins/authorization-plugin/libs/helpers/check-white-route.helper.ts:1]: Yes

**Review by fshabanov — APPROVED** [review]

**Pull request merged**